presentation/src/main/java/com/project/ddoreum/home/HomeViewModel.kt

```diff
@@ -99,24 +116,52 @@ class HomeViewModel @Inject constructor(
     }
 
     init {
-        _name.value = "산또오름"
         _userInfo.value = UserInfo.default
 
-        // TODO : 더미데이터 삭제
-        _recommendList.value = MountainRecommend.defaultList
         _recommendChallengeList.value = ChallengeInfoData.default
         getInProgressChallengeList()
     }
 
+    fun getUserName() {
+        viewModelScope.launch(ioDispatcher) {
+            getUserInfoUseCase.invoke(Unit).onEach { userInfo ->
+                if (userInfo.first.isNotEmpty()) {
+                    _name.emit(userInfo.first)
+                } else {
+                    _name.emit("산또오름")
+                }
+                _state.emit(HomeState.UserInfo)
+            }.launchIn(viewModelScope)
+        }
+    }
+
+    fun getRecommendedMountainData() {
+        viewModelScope.launch(ioDispatcher) {
+            getAllMountainInfoUseCase.invoke(Unit).onEach {
+                it?.filter {
+                    !it.mountainInfo.mountainImage.isNullOrBlank()
+                            && !it.mountainInfo.subTitle.isNullOrBlank()
+                }?.let {
+                    val randomList: ArrayList<MountainInfoData> = arrayListOf()
+                    for (i in 0..2) {
+                        randomList.add(it.random())
+                    }
+
+                    _recommendedMountainList.emit(randomList)
```

Comment:
```suggestion
val randomList = it.shuffled().subList(0,3)
                    _recommendedMountainList.emit(randomList)
```

랜덤하게 산 정보를 가져오는 것 같은데 위와 같은 방식이면 어떨까 하고 제 생각을 적어봅니다.

Reply:
너무 좋습니다! 적용하겠습니다!